### DISCUSSION TITLE
Mounting an object with a background

### DISCUSSION THREAD

**Discussion by rA9Deviants** [lllyasviel/Fooocus#3715] [Ideas]
I need a feature that I can place a png object on a background, and the program merged the object with the background by drawing a shadow.
For example, I have pictures of the kitchen and I need to place a glass on the table and I want the magic of montage to happen.

similar to "Perfect Blend photoshop"

**Comment by Rogala** [comment]
[How to Inpaint In Fooocus](https://www.youtube.com/results?search_query=Fooocus+Inpainting)

**Reply by rA9Deviants** [reply]
That doesn't work for me. I need to put my object on the photo, that the generation would do the editing, draw a shadow. That is, I have a photo of the background, and png of the object, they need to combine naturally

**Reply by Rogala** [reply]
Adobe Photoshop

**Reply by rA9Deviants** [reply]
> Adobe Photoshop

Why are you writing to me if you don’t understand anything? I don't need stupid advice

**Reply by Rogala** [reply]
The magic of montage is to turn to Gandalf.
Mydak

**Reply by rA9Deviants** [reply]
> The magic of montage is to turn to Gandalf. Mydak

I don't understand why you're ruining my post. I'm looking for a neural network function that can do montage. I work with goods, and I have to montage goods on backgrounds, draw shadows, do color correction, and make everything look natural. I want to simplify my work so that I can put my PNG object on the background, and the neural network would do the editing, combining the object with the background.

Why are you telling me about Photoshop, which I've been working with for 15 years? Or are you just a stupid troll?

**Reply by IPv6** [reply]
> Why are you telling me about Photoshop, which I've been working with for 15 years? Or are you just a stupid troll?

Because what you want is not really possible without Photoshop in most cases. At least with SDXL and Fooocus.
So - inpaint, use Photoshop, inpaint again, prepare images for ControlNets as shown in many tutors, iterate, etc.
There are no one-click solution for all problems

**Reply by Rogala** [reply]
You've come to the page of a software product that deals with [image generation](https://github.com/lllyasviel/Fooocus?tab=readme-ov-file#fooocus), not editing. And you want to get a miracle without reading in 2 lines about its real work and possibilities.
1. If you want to work with real images, edit them, combine, etc. for this serves as a graphic editor similar to [Photoshop](https://en.wikipedia.org/wiki/Adobe_Photoshop).

2. If you use Fooocus, you will get generated images based on several possibilities of this system.
A. Create a picture from a text description.
B. Create a drawing analog from a real image with some modifications.
C. Create a picture analogy with a combination of the real picture and the text description.
D. Create a combination of 2-4 pictures that are remotely similar to the original.

3. To these standard features, there are additional features that can modify parts of the drawing by adding or modifying them, such as Inpainting and Outpainting
A. Inpainting can place on the real drawing an object that the AI has generated relative to its description, having previously allocated space for it in the drawing.
B. Inpainting can place on the real drawing the background that the AI generated around the real object in the figure.
С. Outpainting allows to enlarge the picture with the generation of a new empty space relative to the composition in the picture

So the answer to whether Fooocus can merge two real drawings - no, for this you should use a graphic editor, which you said you know how to use.

I'm sorry I didn't write you all this at once and led to confusion and misunderstanding.

**Reply by rA9Deviants** [reply]
> You've come to the page of a software product that deals with [image generation](https://github.com/lllyasviel/Fooocus?tab=readme-ov-file#fooocus), not editing. And you want to get a miracle without reading in 2 lines about its real work and possibilities.
> 
> 1. If you want to work with real images, edit them, combine, etc. for this serves as a graphic editor similar to [Photoshop](https://en.wikipedia.org/wiki/Adobe_Photoshop).
> 2. If you use Fooocus, you will get generated images based on several possibilities of this system.
>    A. Create a picture from a text description.
>    B. Create a drawing analog from a real image with some modifications.
>    C. Create a picture analogy with a combination of the real picture and the text description.
>    D. Create a combination of 2-4 pictures that are remotely similar to the original.
> 3. To these standard features, there are additional features that can modify parts of the drawing by adding or modifying them, such as Inpainting and Outpainting
>    A. Inpainting can place on the real drawing an object that the AI has generated relative to its description, having previously allocated space for it in the drawing.
>    B. Inpainting can place on the real drawing the background that the AI generated around the real object in the figure.
>    С. Outpainting allows to enlarge the picture with the generation of a new empty space relative to the composition in the picture
> 
> So the answer to whether Fooocus can merge two real drawings - no, for this you should use a graphic editor, which you said you know how to use.
> 
> I'm sorry I didn't write you all this at once and led to confusion and misunderstanding.

you don't understand me, and you don't understand yourself either. I've already explained a lot that I'm looking for a feature that will do editing. This post is in the “Ideas” category and I am writing my idea here. Why do you write me nonsense that I don't understand?

**Reply by Rogala** [reply]
No developer, no one to read the ideas.

**Reply by rA9Deviants** [reply]
> No developer, no one to read the ideas.


I didn't write about editing photos. I am not interested in photo editing. Please be careful when reading. If you don't understand, don't post.

**Reply by rA9Deviants** [reply]
> No developer, no one to read the ideas.

I don't see the point of your answers. You're off-topic
